android/app/src/main/java/app/organicmaps/widget/placepage/PlaceDescriptionFragment.java

```diff
@@ -16,33 +19,161 @@
-    return root;
-  }
-}
+public class PlaceDescriptionFragment extends BaseMwmFragment {
+    public static final String EXTRA_DESCRIPTION = "description";
+    private String getSourceSuffix() {
+        return "<p>" + getString(R.string.article_from_wikipedia) + "</p>";
+    }
+    
+    @SuppressWarnings("NullableProblems")
+    @NonNull
+    private String mDescription;
+    private WebView webView;
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        mDescription = Objects.requireNonNull(requireArguments()
+                .getString(EXTRA_DESCRIPTION));
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
+                              @Nullable Bundle savedInstanceState) {
+        View root = inflater.inflate(R.layout.fragment_place_description, container, false);
+        webView = root.findViewById(R.id.webview);
+        
+        setupWebView();
+        loadDescriptionWithAdaptiveStyling();
+        
+        ViewCompat.setOnApplyWindowInsetsListener(root, WindowInsetUtils.PaddingInsetsListener.excludeTop());
+        
+        return root;
+    }
+
+    private void setupWebView() {
+        webView.setVerticalScrollBarEnabled(true);
+        webView.setWebViewClient(new WebViewClient());
+        webView.getSettings().setDefaultTextEncodingName("utf-8");
+        webView.setBackgroundColor(getResources().getColor(android.R.color.transparent));
+    }
+
+    private void loadDescriptionWithAdaptiveStyling() {
+        
+        // Determine current theme (light/dark)
+        int nightModeFlags = getResources().getConfiguration().uiMode & Configuration.UI_MODE_NIGHT_MASK;
+        boolean isDarkMode = nightModeFlags == Configuration.UI_MODE_NIGHT_YES;
+
+        // Adaptive color scheme
+        String textColor = String.format("#%06X", (0xFFFFFF & ContextCompat.getColor(requireContext(), 
+            isDarkMode ? R.color.text_light : R.color.text_dark)));
+        
+        String backgroundColor = String.format("#%06X", (0xFFFFFF & ContextCompat.getColor(requireContext(), 
+            isDarkMode ? R.color.bg_window_night : R.color.bg_window)));
+        
+        String headingColor = String.format("#%06X", (0xFFFFFF & ContextCompat.getColor(requireContext(), 
+            isDarkMode ? R.color.text_light : R.color.text_dark)));
+        
+        String linkColor = String.format("#%06X", (0xFFFFFF & ContextCompat.getColor(requireContext(), 
+            isDarkMode ? R.color.base_accent_night : R.color.base_accent)));
+
+        // Comprehensive HTML and CSS
+        String htmlContent = "<!DOCTYPE html>" +
+            "<html>" +
+            "<head>" +
+            "<meta charset='utf-8'>" +
+            "<meta name='viewport' content='width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no'>" +
+            "<style>" +
```

Comment:
Is there a way to have a separate wikipedia.css file in android assets and load it in this webview (e.g. by `file:///android_asset/wikipedia.css`, or wikipedia_light.css + wikipedia_dark.css if necessary).

Also, does css dark/light theming work with Android 5+? https://stackoverflow.com/a/52986538